src/lib/models/authors.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+import mongoose from 'mongoose';
+import Books from './books.js';
+const Schema = mongoose.Schema;
+
+const authorSchema = Schema({
+  name: {type: String, required: true},
+  age: {type: Number, required: true},
+  numberOfBooks: {type: Number, required: true},
+  booksList:[{type: Schema.Types.ObjectId, refs: 'Books', required: true}], 

```

Comment:
Since you've required this field I cannot create an Author without first creating a book. However, you have a require on the AuthorId in your Books, so I cannot create a Book without having an Author...bit of a deadlock.

You need to remove a requirement from one of these models, otherwise it is impossible to use either one.